text/0691-deprecate-class-binding-and-class-name-bindings.md

```diff
@@ -0,0 +1,162 @@
+```
+
+Given these alternatives, it doesn't make sense any longer to continue
+supporting this transform, _especially_ given that it also affects components
+which do not even support `classNameBindings`, such as Glimmer components. Since
+`class` in this case is an _argument_ and not an attribute, and since this only
+affects curly components, it is unlikely that anyone is actually relying on this
+behavior, but removing this incongruity seems like the best course of action.
+
+## Transition Path
+
+In general, users should convert to use angle bracket syntax to invoke their
+component, and then use standard interpolation within the `class` attribute of
+the component. In cases where this is not possible/desired, users can continue
+using curly invocation with `(concat)` to pass the value.
+
+## How We Teach This
+
+### Deprecation Guide
+
+`classBinding` and `classNameBindings` can currently be passed as arguments to
+components that are invoked with curly invocation. These allow users to
+conditionally bind values to the `class` argument using a microsyntax similar to
+the one that can be defined in a Classic component's class body:
+
+```js
+import Component from '@ember/component';
+
+export default Component.extend({
+  classNameBindings: ['isValid:is-valid:is-invalid']
+});
+```
+
+```hbs
+{{my-component classNameBindings="isValid:is-valid:is-invalid"}}
+```
+
+Each binding is a triplet separated by colons. The first identifier in the
+triplet is the value that the class name should be bound to, the second
+identifier is the name of the class to add if the bound value is truthy, and the
+third value is the name to bind if the value is falsy.
+
+These bindings are additive - they add to the existing bindings that are on the
+class, rather than replacing them. Multiple bindings can also be passed in by
+separating them with a space:
+
+```hbs
+{{my-component
+  classBinding="foo:bar"
+  classNameBindings="some.boundProperty isValid:is-valid:is-invalid"
+}}
+```
+
+
+These bindings can be converted into passing a concatenated string into the
+class argument of the component, using inline `if` to reproduce the same
+behavior. This is most conveniently done by converting the component to use
+angle-bracket invocation at the same time.
+
+Before:
+
+```hbs
+{{my-component
+  classBinding="foo:bar"
+  classNameBindings="some.boundProperty isValid:is-valid:is-invalid"
+}}
+```
+
+After:
+
+```hbs
+<MyComponent
+  @class="
+    {{if this.foo "bar"}}
+    {{if this.some.boundProperty "bound-property"}}
+    {{if this.isValid "is-valid" "is-invalid"}}
+  "
+>
+```
+
+Note that we are passing in the `@class` argument, not the `class` attribute.
+This is to maintain the existing behavior exactly, but in most cases it will be
+safe to convert to using the `class` attribute as well. It is only unsafe if you
+were directly referencing the value of the `@class` argument anywhere in your
+component.
```

Comment:
I suggest to flip this around and use the `class=` syntax in the example (which is what most people are going to look at and copy) and only mention the caveat in text. I think it's extremely unlikely that this will matter to real apps – you would have to know that `classBinding` and `classNameBindings` transforms into the `class` argument, plus it doesn't give you access to other classes added by the JS code. You could argue this is not really supported even.